src/components/HistoryList/items/HistoryListItem.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2021 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import * as React from 'react';
+import styled from 'styled-components/native';
+
+// Components
+import Icon, { type IconName } from 'components/modern/Icon';
+import Image from 'components/Image';
+import Text from 'components/modern/Text';
+import TransactionStatusIcon from 'components/modern/TransactionStatusIcon';
+
+// Utils
+import { useThemeColors } from 'utils/themes';
+import { spacing } from 'utils/variables';
+
+// Types
+import type { TransactionStatus } from 'models/History';
+
+type Props = {|
+  title: ?string,
+  subtitle?: ?string,
+  iconUrl?: ?string,
+  iconName?: IconName,
+  iconColor?: string,
+  iconBorderColor?: string,
+  iconComponent?: React.Node,
+  valueComponent?: React.Node,
+  status?: TransactionStatus,
+  onPress?: () => mixed,
+|};
+
+function HistoryListItem({
+  title,
+  subtitle,
+  iconUrl,
+  iconName,
+  iconColor,
+  iconBorderColor,
+  iconComponent,
+  valueComponent,
+  status,
+  onPress,
+}: Props) {
+  const colors = useThemeColors();
+
+  return (
+    <TouchableContainer onPress={onPress} disabled={!onPress}>
+      <LeftColumn>
```

Comment:
Is this ok if there's no `iconUrl`, `iconName` and no `iconComponent` to leave it as it is since it occupies defined with alone?

Reply:
IMO it makes sense, because this is a list item and either we block the place or move rest of the element around. Blocking the space IMO seems better solution.